armory/metrics/task.py

```diff
@@ -1645,32 +1652,34 @@ def preprocess(self, gt_data, tracker_data, tracked_class):
             num_gt_dets += len(data["gt_ids"][t])
 
         # Re-label IDs such that there are no empty IDs
-        if len(unique_gt_ids) > 0:
-            unique_gt_ids = np.unique(unique_gt_ids)
-            gt_id_map = np.nan * np.ones((np.max(unique_gt_ids) + 1))
-            gt_id_map[unique_gt_ids] = np.arange(len(unique_gt_ids))
-            for t in range(num_timesteps):
-                if len(data["gt_ids"][t]) > 0:
-                    data["gt_ids"][t] = gt_id_map[data["gt_ids"][t]].astype(int)
-        if len(unique_tracker_ids) > 0:
-            unique_tracker_ids = np.unique(unique_tracker_ids)
-            tracker_id_map = np.nan * np.ones((np.max(unique_tracker_ids) + 1))
-            tracker_id_map[unique_tracker_ids] = np.arange(len(unique_tracker_ids))
-            for t in range(num_timesteps):
-                if len(data["tracker_ids"][t]) > 0:
-                    data["tracker_ids"][t] = tracker_id_map[
-                        data["tracker_ids"][t]
-                    ].astype(int)
+        def relabel(unique_ids, num_timesteps, sub_data):
```

Comment:
it's a little difficult to grok what's going on with this function. If there are any comments or variable renaming that you think could help with this, that'd be nice to add but otherwise it appears the function is doing what it's supposed to be

Reply:
I think the issue was the excessive use of numpy when unnecessary. Take a look now.

Reply:
looks a lot more readable now